x/nexus/keeper/grpc_query.go

```diff
@@ -150,18 +150,13 @@ func (q Querier) TransferFee(c context.Context, req *types.TransferFeeRequest) (
 func (q Querier) Chains(c context.Context, req *types.ChainsRequest) (*types.ChainsResponse, error) {
 	ctx := sdk.UnwrapSDKContext(c)
 
-	chains := q.keeper.GetChains(ctx)
-
-	switch req.Status {
-	case types.Activated:
-		chains = slices.Filter(chains, func(chain nexus.Chain) bool { return q.keeper.IsChainActivated(ctx, chain) })
-	case types.Deactivated:
-		chains = slices.Filter(chains, func(chain nexus.Chain) bool { return !q.keeper.IsChainActivated(ctx, chain) })
+	if req.Module != "" && req.Module != axelarnet.ModuleName && req.Module != evm.ModuleName {
```

Comment:
this is adding more coupling for better error reporting (unlike evm, axelarnet module value could be a bit confusing and people can get it wrong). While we're unlikely to add more module types (except wasm maybe), open to removing this check.

Reply:
You don't need to do this after making it an enum.